api/filters/authn/oauth/authenticator_test.go

```diff
@@ -313,90 +311,135 @@ var _ = Describe("OIDC Authenticator", func() {
 
-						fakeToken = &webfakes.FakeTokenData{}
-						fakeToken.ClaimsReturns(expectedError)
+							fakeToken = &webfakes.FakeTokenData{}
+							fakeToken.ClaimsReturns(expectedError)
 
-						verifier.VerifyReturns(fakeToken, nil)
+							verifier.VerifyReturns(fakeToken, nil)
 
-					})
+						})
 
-					It("should deny with an error", func() {
-						user, decision, err := authenticator.Authenticate(request)
+						It("should deny with an error", func() {
+							user, decision, err := authenticator.Authenticate(request)
 
-						Expect(user).To(BeNil())
-						Expect(decision).To(Equal(security.Deny))
-						Expect(err).To(Equal(expectedError))
+							Expect(user).To(BeNil())
+							Expect(decision).To(Equal(security.Deny))
+							Expect(err).To(Equal(expectedError))
+						})
 					})
-				})
-
-				Context("when returned token is valid", func() {
-					expectedUserName := "test_user"
 
-					BeforeEach(func() {
-						tokenJSON := fmt.Sprintf(`{"user_name": "%s", "abc": "xyz"}`, expectedUserName)
-						token := &webfakes.FakeTokenData{}
-						token.ClaimsStub = func(v interface{}) error {
-							return json.Unmarshal([]byte(tokenJSON), v)
-						}
-						verifier.VerifyReturns(token, nil)
+					Context("when returned token is valid", func() {
+						expectedUserName := "test_user"
+
+						BeforeEach(func() {
+							tokenJSON := fmt.Sprintf(`{"user_name": "%s", "abc": "xyz"}`, expectedUserName)
+							token := &webfakes.FakeTokenData{}
+							token.ClaimsStub = func(v interface{}) error {
+								return json.Unmarshal([]byte(tokenJSON), v)
+							}
+							verifier.VerifyReturns(token, nil)
+						})
+
+						It("should allow authentication and return user", func() {
+							user, decision, err := authenticator.Authenticate(request)
+
+							Expect(user).To(Not(BeNil()))
+							Expect(user.Name).To(Equal(expectedUserName))
+							Expect(decision).To(Equal(security.Allow))
+							Expect(err).To(BeNil())
+
+							claims := struct {
+								Abc string
+							}{}
+							err = user.Claims(&claims)
+							Expect(err).To(BeNil())
+							Expect(claims.Abc).To(Equal("xyz"))
+						})
 					})
+				})
 
-					It("should allow authentication and return user", func() {
-						user, decision, err := authenticator.Authenticate(request)
+				Context("when bearer is lowercase", func() {
+					BeforeEach(func() {
+						verifier = &securityfakes.FakeTokenVerifier{}
+						authenticator = &oauthAuthenticator{Verifier: verifier}
 
-						Expect(user).To(Not(BeNil()))
-						Expect(user.Name).To(Equal(expectedUserName))
-						Expect(decision).To(Equal(security.Allow))
-						Expect(err).To(BeNil())
+						request.Header.Set("Authorization", "bearer token")
+					})
 
-						claims := struct {
-							Abc string
-						}{}
-						err = user.Claims(&claims)
-						Expect(err).To(BeNil())
-						Expect(claims.Abc).To(Equal("xyz"))
+					Context("when returned token is valid", func() {
```

Comment:
this seems pretty similar (duplicates) the test above. I don't insist but if you feel like, you could parameterize the tests (so you could cover all 3 scenarios for both lowercase and uppercase and avoid the code dup

Reply:
Good catch, I forgot that I rearranged the tests.